lib/scala/kg/src/main/scala/models/graphql/AbstractKgGraphQlSchemaDefinition.scala

```diff
@@ -51,22 +51,37 @@ abstract class AbstractKgGraphQlSchemaDefinition extends BaseGraphQlSchemaDefini
     )
   )
 
+  // Input enum types
+  implicit val KgNodeFieldType = deriveEnumType[KgNodeField.Value](
```

Comment:
If you use circe + enumeratum this is easier. Here's an example enum definition:

```
import enumeratum.values.{StringCirceEnum, StringEnum, StringEnumEntry}
import sangria.macros.derive.deriveEnumType

sealed abstract class StepOrderFlag(val value: String) extends StringEnumEntry

case object StepOrderFlag extends StringEnum[StepOrderFlag] with StringCirceEnum[StepOrderFlag] {
  case object Causal extends StepOrderFlag("causal")
  case object Optional extends StepOrderFlag("optional")
  case object Precondition extends StepOrderFlag("precondition")
  case object Simultaneous extends StepOrderFlag("simultaneous")
  case object Subevent extends StepOrderFlag("subevent")
  val sangriaType = deriveEnumType[StepOrderFlag]()
  val values = findValues
}
```

You can also use `sealed trait`.

Note that the deriveEnumType is there. You can pull it in as an implicit val = SomeEnum.sangriaType. The derive has to be within the case object in order to pick up all the values. Don't ask me why.

Independently of GraphQL, this style of enums (sealed traits + case objects inside another object) is preferred over Enumeration in Scala.

Reply:
Interesting, this is neat, will change